### DISCUSSION TITLE
I have an error

### DISCUSSION THREAD

**Discussion by exterminateur01** [LOENS2/pterodactyl_exporter#35] [Q&A] [answered]
Hello, I have an error with Pterodactyl Exporter, when I put in **Owner**, everything works well, but when I put **Admin-all**, I have an error (see in the screen). (I use the pterodactyl egg)![screen error](https://github.com/user-attachments/assets/69138536-453a-49d9-8e24-5f8494b51324)

**Comment by LOENS2** [comment]
This seemingly has to do with backup permissions. Are you sure that your account has sufficient privileges to access all servers' backups?

**Comment by exterminateur01** [comment]
I am an administrator on the pterodactyl

**Comment by exterminateur01** [comment]
Add me on discord (exterminateur_07)

**Comment by LOENS2** [comment]
I'd rather keep this on GitHub, so that the solution to the problem is transparent to others and for personal reasons (I'm trying to move away from Discord due to their weird decisions in the recent past.) I'm already working on fixing this problem, but I don't really understand it yet. The only time the `max()` function is used is in `http_client.py` line 83. I don't understand how it can be an empty sequence but not null at the same time.

**Comment by exterminateur01** [comment]
So how do I fix this? Because I need it for my business

If I delete this sequence it should work?

**Comment by LOENS2** [comment, marked as answer]
The problem is that I don't know what circumstances lead to this behaviour. You could try changing lines 83-87 in `http_client.py` to:
```
        last_successful_backup = max(
            (dateutil.parser.isoparse(backup["attributes"]["completed_at"]).timestamp()
            for backup in backups
            if backup["attributes"]["is_successful"]), default=0
        ) if backups else 0
```

This introduces a default value to the max() function in case the sequence is empty.

**Comment by LOENS2** [comment]
The only way this could happen is if a server has only failed backups. This is the only case that is not covered here. If that is the case, just manually back up that server and it should be fine again. I haven't considered that niche case so there was no way that this problem would be resolved.

**Comment by LOENS2** [comment]
I have to add that case to my unit tests in order to prevent such failure in the future.

**Comment by exterminateur01** [comment]
Nice it works, is it possible to recover the number of players? I have the PlayerCounter addon

**Comment by exterminateur01** [comment]
And also, possible when searching the server with the server ID (the first part of the ID

**Comment by LOENS2** [comment]
Unfortunately, I can and will not provide support for proprietary and paid addons, as there no commonly accessible documentation on how the provider of this plugin integrates into the API and I don't want to get in any legal trouble either. I aim to provide a metrics exporter for the standard installation of the Pterodactyl Panel. I'm sorry.
But feel free to fork my project and add the functionality you want.

**Comment by exterminateur01** [comment]
Ok And also, possible when searching the server with the server ID (the first part of the ID
